src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionMissingOverrideTest.java

```diff
@@ -0,0 +1,198 @@
+
+        );
+
+        runVerifications(moduleConfig, fileToProcess, expectedViolation,
+                expectedXpathQueries);
+
+    }
+
+    @Test
+    public void testInheritDocInvalid1() throws Exception {
+        final File fileToProcess = new File(getPath(
+                "SuppressionXpathRegressionMissingOverrideInheritDocInvalid1.java"));
+
+        final DefaultConfiguration moduleConfig =
+                createModuleConfig(MissingOverrideCheck.class);
+
+        final String[] expectedViolation = {
+            "7:5: " + getCheckMessage(MissingOverrideCheck.class,
+                MissingOverrideCheck.MSG_KEY_TAG_NOT_VALID_ON, "{@inheritDoc}"),
+        };
+
+        final List<String> expectedXpathQueries = Arrays.asList(
+                "/CLASS_DEF[./IDENT"
+                        + "[@text='SuppressionXpathRegressionMissingOverrideInheritDocInvalid1']]"
+                        + "/OBJBLOCK/METHOD_DEF[./IDENT[@text='test']]",
+                "/CLASS_DEF[./IDENT"
+                        + "[@text='SuppressionXpathRegressionMissingOverrideInheritDocInvalid1']]"
+                        + "/OBJBLOCK/METHOD_DEF[./IDENT[@text='test']]/MODIFIERS",
+                "/CLASS_DEF[./IDENT"
+                        + "[@text='SuppressionXpathRegressionMissingOverrideInheritDocInvalid1']]"
+                        + "/OBJBLOCK/METHOD_DEF[./IDENT[@text='test']]/MODIFIERS/LITERAL_PRIVATE"
+
+        );
+
+        runVerifications(moduleConfig, fileToProcess, expectedViolation,
+                expectedXpathQueries);
+
+    }
+
+    @Test
+    public void testInheritDocInvalid2() throws Exception {
+        final File fileToProcess = new File(getPath(
+                "SuppressionXpathRegressionMissingOverrideInheritDocInvalid2.java"));
+
+        final DefaultConfiguration moduleConfig =
+                createModuleConfig(MissingOverrideCheck.class);
+
+        final String[] expectedViolation = {
+            "7:5: " + getCheckMessage(MissingOverrideCheck.class,
+                MissingOverrideCheck.MSG_KEY_TAG_NOT_VALID_ON, "{@inheritDoc}"),
+        };
+
+        final List<String> expectedXpathQueries = Arrays.asList(
+                "/CLASS_DEF[./IDENT"
+                        + "[@text='SuppressionXpathRegressionMissingOverrideInheritDocInvalid2']]"
+                        + "/OBJBLOCK/METHOD_DEF[./IDENT[@text='test']]",
+                "/CLASS_DEF[./IDENT"
+                        + "[@text='SuppressionXpathRegressionMissingOverrideInheritDocInvalid2']]"
+                        + "/OBJBLOCK/METHOD_DEF[./IDENT[@text='test']]/MODIFIERS",
+                "/CLASS_DEF[./IDENT"
+                        + "[@text='SuppressionXpathRegressionMissingOverrideInheritDocInvalid2']]"
+                        + "/OBJBLOCK/METHOD_DEF[./IDENT[@text='test']]/MODIFIERS/LITERAL_PUBLIC"
+
+        );
+
+        runVerifications(moduleConfig, fileToProcess, expectedViolation,
+                expectedXpathQueries);
+
+    }
```

Comment:
It would be nice to have test with `javaFiveCompatibility` property set to true

Reply:
done